QUALITY_DECLARATION.md

```diff
@@ -0,0 +1,199 @@
+Currently due to limited support on Windows there are some exclusions for testing on the CI.
+
+### Feature Testing [4.i]
+
+The features of iceoryx are tested by the [iceoryx_integrationtest](https://github.com/eclipse-iceoryx/iceoryx/tree/master/iceoryx_integrationtest) package. Using `launch_test` are tests on system-level executed to ensure that the iceoryx packages are functional and the Public API usable.
+There is continuos effort ongoing to cover the corner-cases in the usage of iceoryx.
+
+### Public API Testing [4.ii]
+
+For every major feature in iceoryx are tests executed. It is mandatory for new features to provide tests that covers the code changes in the Pull-Request. The tests reside in separated folders for every package following a defined structure and naming convention.
+The features are tested on module(unit) -integration and system test level. [Guidelines](https://iceoryx.io/latest/advanced/best-practice-for-testing/) for Contributors ensure a high quality for test development.
+
+### Coverage [4.iii]
+
+The Eclipse iceoryx project use [gcov](https://gcc.gnu.org/onlinedocs/gcc/Gcov.html) to capture the following coverage types:
+
+- Line Coverage
+- Function Coverage
+- Branch Coverage
+
+The coverage results of every Pull-Request and master branch are publicly available in [codecov.io](https://app.codecov.io/gh/eclipse-iceoryx/iceoryx).
+If the user want a more detailed report (e.g. the coverage in the different packages), he can follow [this](https://github.com/eclipse-iceoryx/iceoryx/blob/master/CONTRIBUTING.md#coverage-scan) guide to obtain them locally.
+
+### Performance [4.iv]
+
+The most important measurement units for performance testing on iceoryx are the data transfer latency and the CPU load.
+They can be obtained in two ways:
+
+1. [iceperf](https://github.com/eclipse-iceoryx/iceoryx/tree/master/iceoryx_examples/iceperf#iceperf)
+This performance test uses directly the iceoryx API without any indirections to compare different IPC mechanisms on increasing payload sizes. The output is a table linking the payload sizes with the latency in microseconds.
+
+2. [performance_test](https://gitlab.com/ApexAI/performance_test/-/tree/master/)
+The `performance_test` package tests various middleware implementations in ROS using inter-process communication.
+It offers a communication binding for iceoryx and gives information about latency, CPU usage etc.
+Additionally it is possible to test here the integration of iceoryx + CycloneDDS in ROS 2 using the rmw communication abstraction.
+
+Currently there are no automatic regression tests in the CI running, the work to integrate this is addressed in [https://github.com/eclipse-iceoryx/iceoryx/issues/137](https://github.com/eclipse-iceoryx/iceoryx/issues/137). The performance tests are currently executed manually before every release that will integrated into ROS by the maintainers to avoid performance regression.
```

Comment:
```suggestion
Currently there are no automatic regression tests in the CI running, the work to integrate this is addressed in [https://github.com/eclipse-iceoryx/iceoryx/issues/137](https://github.com/eclipse-iceoryx/iceoryx/issues/137). The performance tests are currently executed manually before every release that will be integrated into ROS by the maintainers to avoid performance regression.
```